Add scheduler.alpha.kubernetes.io/critical-pod annotation (#38272)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Refer to [Converting the volume mode of a Snapshot](/docs/concepts/storage/volume-snapshots/#convert-volume-mode)
and the [Kubernetes CSI Developer Documentation](https://kubernetes-csi.github.io/docs/) for more information.

### scheduler.alpha.kubernetes.io/critical-pod (deprecated)

Example: `scheduler.alpha.kubernetes.io/critical-pod: ""`

Used on: Pod

This annotation lets Kubernetes control plane know about a pod being a critical pod so that the descheduler will not remove this pod.

> {{< note >}} Starting in v1.16, this label is removed in favor of [https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/](#Pod Priority). {{< /note >}}

Reviewer: ```suggestion
{{< note >}} Starting in v1.16, this annotation is removed in favor of [Pod Priority](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/). {{< /note >}}
```

Reviewer 2: Even better IMO:
```suggestion
{{< note >}} Starting in v1.16, this annotation was removed in favor of [Pod Priority](/docs/concepts/scheduling-eviction/pod-priority-preemption/). {{< /note >}}
```

Reviewer: Good catch, Tim! Thanks for suggesting this one :)

Author: Pushed the changes, thank you for the suggestions!
